control/iosys.py

```diff
@@ -108,6 +108,14 @@ class for a set of subclasses that are used to implement specific
       The default is to return the entire system state.
 
     """
+
+    idCounter = 0
+    def name_or_default(self, name=None):
+        if name is None:
+            name = "sys[{}]".format(InputOutputSystem.idCounter)
+            InputOutputSystem.idCounter += 1
```

Comment:
Re: conflicting names. Maybe register all system names module or class-wide and check here before assigning. If the name is already taken, just increment the counter once more.

Reply:
I tried implementing this with a __del__ method that would remove an object's name from the class-wide register after this object gets deleted or runs out of scope (to prevent annoying errors being raised when trying to create an object with the same name but in a different scope)

It seems that python's __del__ method isn't really a destructor, and all sorts of bugs happen because of python's reference-counting garbage collector. (https://eli.thegreenplace.net/2009/06/12/safely-using-destructors-in-python)

One potential solution they say is to have users create objects using (https://stackoverflow.com/questions/865115/how-do-i-correctly-clean-up-a-python-object)
`with InputOutputSystem() as sys:
   ...`
But this seems utterly annoying.

Reply:
Or just let the user take care of it. If the user want's to create 100 systems with the same name, fine. Just either throw an error at the time of combining two systems with the same name, or, if you want to be more graceful, rename them into `samename_1` and  `samename_2`.

Reply:
Just made these changes. Also added unit tests that make sure the number of states/inputs/outputs is correct (which requires having renamed all duplicate system names).